nvtabular/tools/dataset_inspector.py

```diff
@@ -0,0 +1,191 @@
+            else:
+                data[col][key_names["mean"][col_type]] = int(ddf[col].mean().compute())
+
+        # For conts get also std
+        if col_type == "cont":
+            data[col]["std"] = ddf[col].std().compute()
+
+        # Get percentage of nan for all
+        data[col]["per_nan"] = 100 * (1 - ddf[col].count().compute() / len(ddf[col]))
+
+    def inspect(self, path, dataset_format, columns_dict, output_file):
+        """
+        Parameters
+        -----------
+        path: str, list of str, or <dask.dataframe|cudf|pd>.DataFrame
+            Dataset path (or list of paths), or a DataFrame. If string,
+            should specify a specific file or directory path. If this is a
+            directory path, the directory structure must be flat (nested
+            directories are not yet supported).
+        dataset_format: string
+            Dataset format (i.e parquet or csv)
+        columns_dict: dictionary
+            Dictionary indicating the diferent columns type
+        output_file: string
+            Filename to write the output statistics
+        """
+        # Get dataset columns
+        cats = columns_dict["cats"]
+        conts = columns_dict["conts"]
+        labels = columns_dict["labels"]
+
+        # Dictionary with json output key names
+        key_names = {}
+        key_names["min"] = {}
+        key_names["min"]["cat"] = "min_entry_size"
+        key_names["min"]["cont"] = "min_val"
+        key_names["max"] = {}
+        key_names["max"]["cat"] = "max_entry_size"
+        key_names["max"]["cont"] = "max_val"
+        key_names["mean"] = {}
+        key_names["mean"]["cat"] = "avg_entry_size"
+        key_names["mean"]["cont"] = "mean"
+
+        # Get dataset
+        dataset = Dataset(path, engine=dataset_format)
+        ddf = dataset.to_ddf()
+
+        # Create Dask Cluster
+        cluster = LocalCUDACluster()
+        client = Client(cluster)
+
+        # Dictionary to store collected information
+        data = {}
+        # Store general info
+        data["num_rows"] = ddf.shape[0].compute()
+
+        # Compute first row to know correct dtypes
+        ddf_dtypes = ddf.head(1)
+
+        # Get continuous columns stats
+        data["conts"] = {}
+        for col in conts:
+            self.__get_stats(ddf, ddf_dtypes, col, data["conts"], "cont", key_names)
+
+        # Get categoricals columns stats
+        data["cats"] = {}
+        for col in cats:
+            self.__get_stats(ddf, ddf_dtypes, col, data["cats"], "cat", key_names)
+
+        # Get labels columns stats
+        data["labels"] = {}
+        for col in labels:
+            self.__get_stats(ddf, ddf_dtypes, col, data["labels"], "label", key_names)
+
+        # Write json file
+        with fsspec.open(output_file, "w") as outfile:
+            json.dump(data, outfile, cls=NpEncoder)
+
+        # Stop Dask Cluster
+        client.shutdown()
+        cluster.close()

```

Comment:
we should use a context manager for this - to make sure we get shutdown/close on exceptions